server/store/memory/peer_store.go

```diff
@@ -271,8 +271,14 @@ func (s *peerStore) AnnouncePeers(infoHash chihaya.InfoHash, seeder bool, numWan
 				}
 
 				if p.IP.To4() == nil {
+					if p.Equal(&peer6) {
+						continue
```

Comment:
Oops -- was reading wayyy too fast. Should stop doing chihaya review during the workday 🙃